Code/GraphMol/SanitException.h

```diff
@@ -26,14 +26,93 @@ namespace RDKit {
 //! class for flagging sanitization errors
 class RDKIT_GRAPHMOL_EXPORT MolSanitizeException : public std::exception {
  public:
-  MolSanitizeException(const char *msg) : _msg(msg){};
-  MolSanitizeException(const std::string &msg) : _msg(msg){};
-  const char *message() const { return _msg.c_str(); };
-  ~MolSanitizeException() throw(){};
+  MolSanitizeException(const char *msg) : d_msg(msg){};
+  MolSanitizeException(const std::string &msg) : d_msg(msg){};
+  MolSanitizeException(const MolSanitizeException &other)
+      : d_msg(other.d_msg){};
+  virtual const char *message() const { return d_msg.c_str(); };
```

Comment:
That would be an API change to existing code, something I am generally reluctant to do.
That doesn’t mean we can’t do it, but we definitely need to think about it carefully

Reply:
That makes sense.

Reply:
I think it might be worth it.  This may explain why some errors aren't propagated to python/jupyter, I think boost python looks for "what" to set the python exception message.

Reply:
It's worth discussing, but I don't think it's relevant to boost.python... we translate exceptions there so that they show up as Python exceptions.

Reply:
Yeah, you are right, I thought boost did this, but I think we do it manually.